File: leetcode/two_sum.py
# 1. 이중 for-loop을 돈다. 
# 2. 돌면서 두 값이 해당 원하는 값과 일치하면 종료한다. 

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for i in range(0, len(nums)):
            for j in range(i+1, len(nums)):
                if nums[i] + nums[j] == target:
                    return [i, j]

File: leetcode/TwoSum.java
import java.util.HashMap;

public class TwoSum {

    /**
     * nums : 배열 안에 숫자들 주어짐
     * target : 두 개의 숫자를 더해서 만들어 야 할 값
     **/
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            // 구해야 할 값
            int complement = target - nums[i];
            // 값이 있으면 map 에서 꺼내 쓰자
            if (map.containsKey(complement)){
                return new int[] { map.get(complement), i};
            } else {
                map.put(nums[i], i);
            }
        }
        return new int[0];
    }
}
